lib/index.js

```diff
@@ -0,0 +1,255 @@
+ * @param {string} options.controller - The ID of the controller of the key.
+ * @param {string} options.keyId - The ID of the signing key to use.
+ * @param {Uint8Array|string} options.data - The data to sign as a Uint8Array
+ *   or a base64url-encoded string.
+ *
+ * @returns {Promise<Object>} An object containing `{signature}`.
+ */
+exports.sign = async ({controller, keyId, data}) => {
+  assert.string(controller, 'options.controller');
+  assert.string(keyId, 'options.keyId');
+  assert.string(data, 'options.data');
+
+  const {key} = await _getKeyRecord({id: keyId, controller});
+
+  const signature = await _hs256Sign({key, data});
+  return {signature: base64url.encode(signature)};
+};
+
+/**
+ * Verifies some data. Note that the data will be sent to the server, so if
+ * this data is intended to be secret it should be hashed first. However,
+ * hashing the data first may present interoperability issues so choose
+ * wisely.
+ *
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The ID of the controller of the key.
+ * @param {string} options.keyId - The ID of the signing key to use.
+ * @param {Uint8Array|string} options.data - The data to sign as a Uint8Array
+ *   or a base64url-encoded string.
+ * @param {string} options.signature - The base64url-encoded signature to
+ *   verify.
+ *
+ * @returns {Promise<Object>} An object containing `{verified}`.
+ */
+exports.verify = async ({controller, keyId, data, signature}) => {
+  assert.string(controller, 'options.controller');
+  assert.string(keyId, 'options.keyId');
+  assert.string(data, 'options.data');
+  assert.string(signature, 'options.signature');
+
+  const {key} = await _getKeyRecord({id: keyId, controller});
+
+  const verified = await _hs256Verify({key, data, signature});
+  return {verified};
+};
+
+/**
+ * Gets a previously stored key record.
+ *
+ * @ignore
+ * @param {Object} options - The options to use.
+ * @param {string} options.controller - The controller of the key.
+ * @param {string} options.id - The ID of the key.
+ *
+ * @returns {Promise<Object>} The key record.
+ */
+async function _getKeyRecord({controller, id}) {
+  assert.string(controller, 'options.controller');
+  assert.string(id, 'options.id');
+
+  const record = await database.collections.ssm.findOne(
+    {controller: database.hash(controller), id: database.hash(id)},
+    {_id: 0, key: 1, meta: 1});
+  if(!record) {
+    throw new BedrockError(
+      'Key not found.',
+      'NotFoundError',
+      {key: id, controller, httpStatusCode: 404, public: true});
+  }
+
+  return record;
+}
+
+async function _aesWrapKey({kek, key}) {
+  key = base64url.decode(key);
+  const secret = base64url.decode(kek.secret);
+  const cipher = crypto.createCipheriv(
+    AES_KW_ALGORITHM, secret, AES_KW_RFC3394_IV);
+  const output = Buffer.concat([cipher.update(key), cipher.final()]);
+  return base64url.encode(output);
+}
+
+async function _aesUnwrapKey({kek, wrappedKey}) {
```

Comment:
I note that most if not all of these crypto helpers are sync code.  Is there a good reason for them to be wrapped in async?

Reply:
I suppose the helpers could be synchronous if we find that helpful ... I wouldn't want the public functions to be synchronous though. I was just trying to be consistent with how I'd expect this stuff to function (which is asynchronously).